cmd/akamai-purger/main.go

```diff
@@ -43,50 +50,50 @@ type Config struct {
 	Beeline cmd.BeelineConfig
 }
 
+// akamaiPurger is a mutex protected container for a gRPC server which receives
+// requests to purge the URLs of OCSP responses cached by Akamai, stores these
+// URLs in an inner slice, and dispatches them to Akamai's Fast Purge API in
+// batches.
 type akamaiPurger struct {
+	sync.Mutex
 	akamaipb.UnimplementedAkamaiPurgerServer
-	mu      sync.Mutex
-	toPurge []string
-
-	client *akamai.CachePurgeClient
-	log    blog.Logger
+	toPurge      []string
+	maxQueueSize int
+	client       *akamai.CachePurgeClient
+	log          blog.Logger
 }
 
 func (ap *akamaiPurger) len() int {
-	ap.mu.Lock()
-	defer ap.mu.Unlock()
+	ap.Lock()
+	defer ap.Unlock()
 	return len(ap.toPurge)
 }
 
 func (ap *akamaiPurger) purge() error {
-	ap.mu.Lock()
+	ap.Lock()
 	urls := ap.toPurge[:]
 	ap.toPurge = []string{}
-	ap.mu.Unlock()
+	ap.Unlock()
 	if len(urls) == 0 {
 		return nil
 	}
 
-	err := ap.client.Purge(urls)
+	stoppedAt, err := ap.client.Purge(urls)
 	if err != nil {
-		// Add the URLs back to the queue.
-		ap.mu.Lock()
-		ap.toPurge = append(urls, ap.toPurge...)
-		ap.mu.Unlock()
+		// Add the remaining URLs back to the queue.
+		ap.Lock()
+		ap.toPurge = append(ap.toPurge, urls[stoppedAt:]...)
```

Comment:
Changing where these URLs are re-added to this toPurge list (they used to be prepended, now they're appended) is a subtle but imo good change. Means that if (somehow) there's a single url-of-death that causes akamai to fail to purge every time, it won't get in the way of other URLs that are being added, it'll only get in the way of its own batch.